### PR TITLE
[Quant][Inductor] expand quantization conv-binary(-unary) pattern fusion inside inductor

### PR DESCRIPTION
**Pull request by jiayisunx**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #138051

### Summary
Expand quantization conv-binary(-unary) pattern fusion inside inductor to support the following two patterns:
Pattern 1:
```
    Conv(X)   extra input
           \   /
            Add
             |
        Optional(relu)
             |
             Y
```
Pattern 2:
```
    extra input   Conv(X)
           \   /
            Add
             |
        Optional(relu)
             |
             Y
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

**Review by leslie-fang-intel — CHANGES_REQUESTED** [review]
Thanks for the PR. Please add the corresponding UT for these patterns.

**Comment by jiayisunx** [comment]
> extra input

Thanks for your suggestion, I have added the corresponding UT for these patterns.

File: test/inductor/test_mkldnn_pattern_matcher.py [hunk 1]
```diff
@@ -940,27 +940,108 @@ def matcher_check_fn():
...
+                self.conv2 = torch.nn.Conv2d(6, 6, kernel_size=3, stride=1, bias=False)
+                self.add_fn2 = add_fn
+                self.relu2 = torch.nn.ReLU()
+                self.use_relu = use_relu
+
+            def forward(self, x, x2, x3):
+                x1 = self.conv1(x)
+                tmp = self.add_fn(x1, x2)
+                if self.use_relu:
+                    tmp = self.relu(tmp)
```
Comment (leslie-fang-intel) [inline test/inductor/test_mkldnn_pattern_matcher.py:989]: Further adding a flag to swap inputs of add?

**Review by leslie-fang-intel — APPROVED** [review]
Looks good to me. I think we should further enhance the `_qconv2d_add_cpu_test_helper2` by swapping inputs of add.

File: test/inductor/test_mkldnn_pattern_matcher.py (hunk 1 above)
Comment (jiayisunx) [reply test/inductor/test_mkldnn_pattern_matcher.py:994]: Added, thanks for your advice!

**Review by jansel — APPROVED** [review]

**Review by jgong5 — APPROVED** [review]
small nits on the code comment, others LGTM.

File: test/inductor/test_mkldnn_pattern_matcher.py [hunk 2]
```diff
@@ -940,27 +940,118 @@ def matcher_check_fn():
                 matcher_check_fn=matcher_check_fn,
             )
 
+    def _qconv2d_add_cpu_test_helper2(self, use_relu=False, int8_mixed_bf16=False):
+        r"""
+        This testcase will quantize two Conv2d->Add patterns as:
+                 X
+               /   \
+        Conv(X)   extra input
```
[the remark this answers is not in the document]
Comment (jiayisunx) [reply test/inductor/test_mkldnn_pattern_matcher.py]: Thanks for your comments! yes, you are right, it's my mistake, I have fixed it.

**Pull request closed without merging**